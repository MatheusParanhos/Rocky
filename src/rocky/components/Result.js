import React, { Component } from "react";
import { Text, View, TouchableOpacity, StyleSheet } from "react-native";
import { deviceWidth, colors, fontFamily } from "../../theme/theme";
import FontAwesome from "react-native-vector-icons/FontAwesome";
import PropTypes from "prop-types";

export default class Result extends Component {
  renderIcon = () => {
    if (this.props.computerChoice === "scissors") {
      return (
        <FontAwesome
          name={`hand-${this.props.computerChoice}-o`}
          size={50}
          color={colors.primary}
          style={{ transform: [{ rotate: "270deg" }] }}
        />
      );
    }
    if (
      this.props.computerChoice === "rock" ||
      this.props.computerChoice === "paper"
    ) {
      return (
        <FontAwesome
          name={`hand-${this.props.computerChoice}-o`}
          size={50}
          color={colors.primary}
          style={{ transform: [{ rotate: "180deg" }] }}
        />
      );
    }
  };
  renderChoice = () => {
    if (this.props.userChoice === "scissors") {
      return (
        <FontAwesome
          name={`hand-${this.props.userChoice}-o`}
          size={50}
          color={colors.primary}
          style={{ transform: [{ rotate: "90deg" }] }}
        />
      );
    }
    if (this.props.userChoice === "rock" || this.props.userChoice === "paper") {
      return (
        <FontAwesome
          name={`hand-${this.props.userChoice}-o`}
          size={50}
          color={colors.primary}
        />
      );
    }
  };
  render() {
    return (
      <View style={styles.container}>
        <View style={styles.iconContainer}>{this.renderIcon()}</View>
        <Text style={styles.text}>{this.props.countdown}</Text>
        <Text style={styles.text}>{this.props.result}</Text>
        <View
          style={styles.choiceContainer}
        >
          {this.renderChoice()}
        </View>
      </View>
    );
  }
}
Result.propTypes = {
  countdown: PropTypes.number,
  result: PropTypes.string,
  userChoice: PropTypes.string,
  computerChoice: PropTypes.string
};
Result.defaultProps = {
  countdown: 0,
  result: "",
  userChoice: "",
  computerChoice: ""
};

const styles = StyleSheet.create({
  container: {
    height: 200,
    width: deviceWidth,
    justifyContent: "center",
    alignItems: "center",
    borderRadius: 25
  },
  text: {
    fontFamily: fontFamily.splash,
    fontSize: 50
  },
  iconContainer: {
    height: 100,
    width: 100,
    alignItems: "center",
    justifyContent: "center"
  },
  choiceContainer:{
    height: 100,
    width: 100,
    marginTop: 30,
    alignItems: "center",
    justifyContent: "center"
  }
});
