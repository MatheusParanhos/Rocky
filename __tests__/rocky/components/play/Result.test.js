import React from "react";
import Result from "../../../../src/rocky/play/components/Result";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";

Enzyme.configure({ adapter: new Adapter() });

test("renders correctly", () => {
  const tree = renderer.create(<Result />).toJSON();
  expect(tree).toMatchSnapshot();
});
it("Testing props", () => {
  const wrapper = shallow(<Result 
    countdown={null}
    result={null}
    userChoice={null}
    computerChoice={null}
    />).props();
  console.warn(wrapper);
  expect(2).toEqual(2);
});
