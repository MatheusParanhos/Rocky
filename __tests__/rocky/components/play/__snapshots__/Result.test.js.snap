// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 25,
      "height": 200,
      "justifyContent": "center",
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 100,
        "justifyContent": "center",
        "width": 100,
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "SignatraDEMO",
        "fontSize": 50,
      }
    }
  >
    0
  </Text>
  <Text
    style={
      Object {
        "fontFamily": "SignatraDEMO",
        "fontSize": 50,
      }
    }
  >
    
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 100,
        "justifyContent": "center",
        "marginTop": 30,
        "width": 100,
      }
    }
  />
</View>
`;
